src/components/IslandsDiagram.astro

```diff
@@ -0,0 +1,79 @@
+---
+function slotDefault (slotName: string) {
+	return `❌ Missing <Fragment slot="${slotName}">...</Fragment>`;
+}
+---
+<div class="diagram">
+	<div class="header app"><slot name="headerApp">{slotDefault('headerApp')}</slot></div>
+	<div class="sidebar app"><slot name="sidebarApp">{slotDefault('sidebarApp')}</slot></div>
+	<div class="main"><slot name="main">{slotDefault('main')}</slot></div>
+	<div class="carousel app"><slot name="carouselApp">{slotDefault('carouselApp')}</slot></div>
+	<div class="adv"><slot name="advertisement">{slotDefault('advertisement')}</slot></div>
+	<div class="footer"><slot name="footer">{slotDefault('footer')}</slot></div>
+</div>
+
+<style>
+	.diagram {
+		display: grid;
+		grid-template-areas:
+			'header header header'
+			'sidebar main main'
+			'sidebar carousel carousel'
+			'adv footer footer';
+		grid-gap: 0.75rem;
+		padding: 0.75rem;
+		background: var(--theme-bg);
+		border: 2px solid var(--theme-divider);
+		overflow-x: auto;
+	}
+	.diagram * {
+		display: flex;
+		justify-content: center;
+		align-items: center;
+		text-align: center;
+		padding: 2rem;
+		background: hsl(0, 0%, 96%);
+		border: 2px solid hsl(0, 0%, 80%);
+		color: hsl(0, 0%, 40%);
+	}
+	.diagram .app {
+		font-size: 1.05rem;
+	}
```

Comment:
I’m not sure this is doing much? I’d either bump up the font size a little more (`1.25rem` looks more like the original image?) or just remove this and have them all be the same size.

Reply:
Yeah, I noticed that too. OK we’ll roll with it!